### ISSUE TITLE
Resnet not using group normalization

### ISSUE THREAD

**Issue by tianboh** [kevinhsieh/non_iid_dml#1]
Hi, I checked the code for resnet in `non_iid_dml/apps/caffe/examples/cifar10/2parts`. However, I didn't see group normalization, which is the key to solve non IID settings. Am I missing anything?
Also, I am trying to replace batch normalization to group normalization. Is there any guideline for assigning group number? In the paper, the author only mentioned that group number is 2 in BN-AlexNet.

**Comment by kevinhsieh** [comment]
You can adapt the `apps/caffe/examples/cifar10/5parts/resnetgn20_train_val.prototxt.template` to have a GroupNorm ResNet-20. You may need to change the learning rates and other hyper-parameters as these settings vary with number of nodes. 

Our experiments show the group number at 2 typically yields best results for ResNet-20. You can adjust the number to find the best setting for your DNN.
